superset/models/core.py

```diff
@@ -930,11 +950,7 @@ def get_df(self, sql, schema, mutator=None):
         username = utils.get_username()
 
         def needs_conversion(df_series):
```

Comment:
While we're at it, perhaps `def needs_conversion(df_series: Series) -> bool` here?